### PR TITLE
Docs: Add plugin dev documentation for logs to trace

### PR DESCRIPTION
**Pull request by ivanahuckova**
This PR adds plugin developer documentation that describes how to implement logs to traces. Currently, this is is possible trough implementing data links:

- url data link 
- internal data link - this is marked as `@internal` so I added it to `## APIs under development`  until we release it.

### CODE REVIEW

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md [hunk 1]
```diff
@@ -197,6 +197,34 @@ const result = new MutableDataFrame({
...
 ```
 
+### Logs to trace using data link with url
+
+If your log data contains **trace IDs**, you can enhance your log data frames by adding a **field with trace ID values** and **URL data links**. These links should utilize the trace ID value to accurately link to the appropriate trace. This enhancement enables users to seamlessly move from log lines to the relevant traces.
+
+**Example in TypeScript:**
+
+```ts
+const result = new MutableDataFrame({
```
Comment (svennergr) [inline docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md:207]: `MutableDataFrame` is deprecated. WDYT about just using `DataFrame`?

```suggestion
const result = new DataFrame({
```

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md [hunk 2]
```diff
@@ -526,6 +554,46 @@ export class ExampleDatasource
...
+
+{{% admonition type="note" %}} This feature is currently not supported for external plugins outside of the Grafana repo. The `@internal` API is currently under development. {{%
+/admonition %}}
+
+If you are developing a data source plugin that handles both logs and traces, and your log data contains **trace IDs**, you can enhance your log data frames by adding a field with **trace ID values** and **internal data links**. These links should utilize the trace ID value to accurately create a trace query that produces relevant trace. This enhancement enables users to seamlessly move from log lines to the traces.
+
+**Example in TypeScript:**
+
+```ts
+const result = new MutableDataFrame({
```
Comment (svennergr) [inline docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md:567]:
```suggestion
const result = new DataFrame({
```

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md [hunk 3]
```diff
@@ -526,6 +554,46 @@ export class ExampleDatasource
...
+        links: [
+          {
+            title: 'Trace view',
+            url: '',
+            internal: {
+              // Be sure to adjust this example with datasourceUid,  datasourceName and query based on your data source logic.
+              datasourceUid: instanceSettings.uid,
+              datasourceName: instanceSettings.name,
+              query: {
+                { ...query, queryType: 'trace', traceId: '${__value.raw}'}; // ${__value.raw} is a variable that will be replaced with actual traceID value.
```
Comment (svennergr) [inline docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md:583]:
```suggestion
                { ...query, queryType: 'trace', traceId: '${__value.raw}'}, // ${__value.raw} is a variable that will be replaced with actual traceID value.
```

**Review by svennergr — COMMENTED** [review]
Left a few comments but the rest LGTM!

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md (hunk 1 above)
Comment (ivanahuckova) [reply docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md]: Good point!  I discovered `createDataFrame` function that will create `dataFrame` and used it in https://github.com/grafana/grafana/pull/73225/commits/d88f23bbd1fba3dfa316b55ef83b9af7c647b610. Other option would be just using `result: DataFrame = {...}`, but then we would need to ensure that `length` is everywhere as it is required field. This handles it for us. Let me know what do you think.

Comment (svennergr) [reply docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md]: I like that! 👍

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md [hunk 4]
```diff
@@ -197,6 +205,36 @@ const result = new MutableDataFrame({
...
 ```
 
+### Logs to trace using data link with url
+
+If your log data contains **trace IDs**, you can enhance your log data frames by adding a **field with trace ID values** and **URL data links**. These links should utilize the trace ID value to accurately link to the appropriate trace. This enhancement enables users to seamlessly move from log lines to the relevant traces.
+
+**Example in TypeScript:**
+
+```ts
+import { createDataFrame } from '@grafana/data';
```
Comment (svennergr) [inline docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md:215]:
```suggestion
import { createDataFrame, FieldType } from '@grafana/data';
```

**Review by svennergr — APPROVED** [review]
LGTM!

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md [hunk 5]
```diff
@@ -197,6 +205,36 @@ const result = new MutableDataFrame({
 });
 ```
 
+### Logs to trace using data link with url
+
+If your log data contains **trace IDs**, you can enhance your log data frames by adding a **field with trace ID values** and **URL data links**. These links should utilize the trace ID value to accurately link to the appropriate trace. This enhancement enables users to seamlessly move from log lines to the relevant traces.
```
Comment (josmperez) [inline docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md:210]:
```suggestion
If your log data contains **trace IDs**, you can enhance your log data frames by adding a field with *trace ID values* and *URL data links*. These links should use the trace ID value to accurately link to the appropriate trace. This enhancement enables users to seamlessly move from log lines to the relevant traces.
```

File: docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md [hunk 6]
```diff
@@ -526,6 +564,48 @@ export class ExampleDatasource
 
 For an example of how to implement the logs sample in the Elasticsearch data source, refer to [PR 70258](https://github.com/grafana/grafana/pull/70258/).
 
+### Logs to trace using internal data links
+
+{{% admonition type="note" %}} This feature is currently not supported for external plugins outside of the Grafana repo. The `@internal` API is currently under development. {{%
+/admonition %}}
+
+If you are developing a data source plugin that handles both logs and traces, and your log data contains **trace IDs**, you can enhance your log data frames by adding a field with **trace ID values** and **internal data links**. These links should utilize the trace ID value to accurately create a trace query that produces relevant trace. This enhancement enables users to seamlessly move from log lines to the traces.
```
Comment (josmperez) [inline docs/sources/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-logs-data-source-plugin.md:572]:
```suggestion
If you are developing a data source plugin that handles both logs and traces, and your log data contains trace IDs, you can enhance your log data frames by adding a field with trace ID values and internal data links. These links should use the trace ID value to accurately create a trace query that produces relevant trace. This enhancement enables users to seamlessly move from log lines to the traces.
```

**Review by josmperez — APPROVED** [review]
A couple minor style tweaks, otherwise LGTM.

**Review by yesoreyeram — APPROVED** [review]
LGTM. Thank you Ivana

**Pull request merged**